src/controllers/controllermanager.cpp

```diff
@@ -174,7 +178,20 @@ void ControllerManager::updateControllerList() {
 
     locker.relock();
     if (newDeviceList != m_controllers) {
+        // Check for keyboard controller
+        KeyboardController* pKeyboardController;
+        foreach (Controller* pController, newDeviceList) {
```

Comment:
Ah nice, I didn't know that there are ranged based loops in C++. ControllerManager has several other usages of the Qt foreach loop. Shall I replace those as well?